### DISCUSSION TITLE
Very Small BirdNET-Pi

### DISCUSSION THREAD

**Discussion by dalogue1** [Nachtzuster/BirdNET-Pi#214] [Show and tell]
I have created a very small BirdNET-Pi using a RPi-Zero 2W and the circuit board from a USB Sound card and a AOM-5024 microphone element.  I made it for a family member and all they will have to do is plug it in outdoors under shelter since it is not waterproof.

I started with a RPi Zero 2W running BirdNET-Pi.  I then removed the circuit board from a cheap $2 USB sound card and soldered the USB connections directly to the test points on the back of the Pi. Finally I connected the microphone connections to a high quality AOM-5024 microphone element.  I used a small case and just a touch of hot glue.

The green led is the power indicator of the Pi and the red LED is the flashing indicator that the USB microphone is being read.

The front and the back of the entire device is shown below:
![image](https://github.com/user-attachments/assets/c75c32a4-97ff-4493-addb-104d3c5c2b5c)

![image](https://github.com/user-attachments/assets/aa153011-b665-4a94-881f-e8dcd6369346)

The connections to the RPi Zero 2W for the USB is explained on this website.
https://othermod.com/raspberry-pi-zero-external-usb-connector/
![image](https://github.com/user-attachments/assets/d5b657d1-1426-41bd-9f13-78830aae2a74)

The microphone element is available on DigiKey.
https://www.digikey.com/en/products/detail/pui-audio-inc/AOM-5024L-HD-R/7898328

The case is available on Amazon.
https://www.amazon.com/dp/B01CQ3UCYM

The cheap USB sound card is easy to find on EBay or AliExpress by searching for "3dsound usb adapter".
(It is the one with the label "3D SOUND" stamped on the front of the case)
I used low melt solder to easily remove the USB connector and the speaker and microphone jacks.
![image](https://github.com/user-attachments/assets/15c0d02b-6237-42b0-9f1c-7b1f9949c9b8)

**Comment by alexbelgium** [comment]
Very nice! Thanks for sharing. What is the performance in terms of time per analysis? On my rpi4 it makes roughly 2/2.5 s per analysis of 9s sample.

**Reply by dalogue1** [reply]
3.5/4s per analysis of 15s sample.

**Reply by dalogue1** [reply]
Not quite as fast as my RPi5 with SSD.  0.68s per analysis of 15s with 1s overlap

**Comment by tphakala** [comment]
That is a very clever way of connecting a USB device to an RPi Zero!

Is the AOM5024 working OK on that sound card? I have tested them with several audio cards, and it didn't work at all on one of those very same-looking cheap ones. I think it was due to too low plug-in power voltage from the sound card.

**Reply by dalogue1** [reply]
You can hear the sound on BirdWeather.com   https://app.birdweather.com/stations/5355

Note: The background hum is my A/C units, annoying but it does not appear to interfere with the detections.  I am currently testing it before sending it away to my family member.

**Reply by danlogue** [reply]
After doing the testing, I realized that the sound card in the USB adapter worked but detected far fewer birds than the better recommended USB adapters.  So I sent the better adapter.

**Reply by brspencer90** [reply]
> After doing the testing, I realized that the sound card in the USB adapter worked but detected far fewer birds than the better recommended USB adapters. So I sent the better adapter.

Which USB soundcard did you use? Did you solder it like this one in this example?

**Reply by dalogue1** [reply]
I used the UGREEN External Sound Card Jack USB Adapter, Model 30724

https://www.amazon.com/dp/B01N905VOY?ref_=ppx_hzsearch_conn_dt_b_fed_asin_title_1&th=1

I did not wire it up like in the image though.  I would like to know if any of the small USB sound card dongles actually work better.
(Tearing apart the UGREEN adapter seems extreme since it cost much more than $2)


The microphone element listed above though has been great.  I have made 4 different microphones using that element.  It works as well as some of the expensive microphones and has great sound pickup.

**Reply by tphakala** [reply]
I have tested cm108 based cards, those with blue bare pcb. I ordered 10pcs from ebay and they were all defective, every single one had huge noise on mic input. I also got few from amazon.de and even those look almost identical they were perfect, they supply high plugin voltage so AOM5024 capsule works well and no noise issue at all.

**Reply by stevedee** [reply]
You may be interested in my mic/USB module testing post: http://captainbodgit.blogspot.com/2024/12/birdnet-pi-usb-audio-module-testing.html 

Main takeaways:-
 - These modules are not designed for BirdNet, so don't expect most expensive units to be best performers.
 - the cm108 is a very capable chip, but performance depends upon how 'system builder' has designed circuit & packaged module.
 - cheap unbranded modules are difficult to 'repurchase' because retailer is probably stocking from a variety of wholesalers.
 
I hope this helps.

**Reply by dalogue1** [reply]
Your analysis of the mic/USB was very interesting.

Currently, my general setup is to use a raw AOM-5024 element such as https://www.digikey.com/en/products/detail/pui-audio-inc/AOM-5024P-HD-MB-R/16585482 attached directly to the microphone input of the USB adapter.

Are there any modifications such as what you show in the below diagram that would potentially improve the quality of the signal fed to the USB adapter?  Since I am already manually constructing the microphone adding an additional resistor or capacitor would be quite simple.  If so, what values should be used? Would a trimmer capacitor or variable resistor allow tuning of the microphone?

Also, is an additional voltage (+V) needed? (Perhaps 5V from a USB port.)


![image](https://github.com/user-attachments/assets/f6b7f5d4-f0ae-48f5-95a6-e11ebf7accf7)

**Reply by stevedee** [reply]
The the AOM-5024P-HD-MB-R is a good choice of Electret microphone capsule.

Putting system & environmental noise to one side for a moment, my post does not imply that improvements in 'quality' are necessary, only that there are variations in sensitivity between various combinations of microphone & USB modules. High sensitivity may be beneficial for BirdNet, as we would generally like to capture as many bird calls as possible within a given area.

Audio/USB adaptor manufacturers do not know which mic we plan to use with their devices. Therefore the 'plug-in-power' (PiP) circuitry may not be optimal for a given combination.
In addition, the vast majority of [non-BirdNet] users require the device to provide good speech sound quality, where the user is very close to the mic. So sensitivity need not be that high.
For BirdNet we probably want to cover a large area (high sensitivity) but without overload (distortion) from sounds close to the mic.
Please see this excellent article which explains the limitations of PiP: https://diymics.com/why-you-shouldnt-rely-on-microphone-plug-in-power/

You may find that it is possible to increase your system sensitivity (if required) by altering the drive voltage & load resistor. Or you could consider adding a preamp between mic & USB module. But is this worth the effort for a bird call sampling system? Especially as you have build such a nice compact unit.

Also, I don't think very high quality audio is necessary. Just consider how BirdNet works;
The system uses spectrograms to analyse the 'shape' & frequency components of the captured audio.
Most bird calls fall in the range 250Hz - 8300Hz. The frequency range of the BirdNet spectrogram has been limited to 150Hz - 15kHz for technical reasons. However I suspect the passband of interest for the microphone audio (if you wanted to apply filtering) could be limited to 250Hz - 12kHz, but this is just my guestimate.

- Note that the excellent application "whoBIRD" (also based upon BirdNet) includes an adjustable high pass filter which allows the LF cutoff to be set in the range 0 to 1000Hz: https://github.com/woheller69/whoBIRD/blob/master/README.md

BirdNet was trained on thousands of recordings (500 max per species) of variable quality & reliability from a number of sources including Xeno-canto & Macaulay. Because of the amount of data involved, manual verification was not practical. 
But they also trained the system with 'non-bird sounds' to guard against false-positives by using recordings of animals, weather, people, vehicles & even recorder noise.
(This could mean that 50 & 60Hz mains hum is less of a problem than I originally thought, but its still nice if you can filter it out).

BirdNet is not a perfect analysis system; it sometimes classifies my neighbours dog as a raven, a local fox as a whooper swan & common doves/pigeons as turtle doves!
But it is a very good system, and we can expect it to get even better in the future.


The point I'm trying to make here is that as long as your system is reasonably sensitive and sounds free of distortion, its probably more than good enough.
As a portable, 'fair-weather' unit it will be limited by where it is 'sited' & how it is used.

**Comment by BAM-Team** [comment]
Excellent job :-D Thanks for shring!

**Comment by BAM-Team** [comment]
I recommend the Boya BY-LM40
I tested more than 10 Mics. It´s the best I found for Birding and with USB connector.

**Comment by DandyDucks** [comment]
Hi there! Is this something you could use with no wifi connection? Will the RPI record all of the "sightings" on it's own until I bring it back and upload them, or would I need to add something else?
I'm asking because I work in the parks dept of a municipality, and we would love to do wildlife surveys but don't have the budget for a PUC. I don't have access to my electronics mentor anymore and want to make sure before I dish out the money towards the project.
Thank you so much!

**Reply by tphakala** [reply]
Audiomoth is what biologists etc. use for usecase like yours https://www.openacousticdevices.info/audiomoth
